designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ScenarioActivityApiHttpService.scala

```diff
@@ -233,9 +228,22 @@ class ScenarioActivityApiHttpService(
           case None =>
             Future.successful(List.empty)
         }
-        combinedActivities       = (generalActivities ++ deploymentManagerSpecificActivities).map(toDto)
-        sortedCombinedActivities = combinedActivities.toList.sortBy(_.date)
-      } yield sortedCombinedActivities
+        combinedActivities = generalActivities ++ deploymentManagerSpecificActivities
+        //  The API endpoint returning scenario activities does not yet have support for filtering. We made a decision to:
+        //  - for activities not related to deployments:        always display them on FE
+        //  - for activities related to batch deployments:      always display them on FE
```

Comment:
Those are the business requirements from task, they can be explicitly represented now, batch deployment error visibility is fixed - those errors should be visible according to docs